src/Engine/ProtoCore/FFI/CLRObjectMarshaler.cs

```diff
@@ -456,20 +458,33 @@ protected T[] UnMarshal<T>(CLRStackValue dsObject, ProtoCore.MSILRuntimeCore run
             var dsElements = dsObject.Value as IList<CLRStackValue>;
```

Comment:
I think I need to add more checks such as this for everywhere that we're doing `stackValue.Value as IList<CLRStackValue>` since it can now be an `IList `of `object`.

Reply:
should this be a helper?

Reply:
This was actually a question to @pinzart90 since I'm not clear if all clr stackvalues that are enumerable are expected to be `IList<CLRStackValue>`.